src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala

```diff
@@ -113,23 +124,231 @@ final case class ThermalHouse(
     ThermalEnergyDemand(requiredEnergy, possibleEnergy)
   }
 
-  /** Calculate the needed energy to change from start temperature to target
-    * temperature.
+  /** Checks if the simulation reached a new full hour in the simulation time.
+    * If we already simulated a tick within the current simulation time hour, it
+    * returns None. Else a sequence of the hours since the tick of state will be
+    * returned.
```

Comment:
Did I understand it correctly that we only check the demand once per hour?

Reply:
Yes. There is only hourly data available so far.